provider/pkg/provider/helm_release.go

```diff
@@ -299,9 +300,11 @@ func decodeRelease(pm resource.PropertyMap, label string) (*Release, error) {
 					if err != nil {
 						return nil, err
 					}
-					if err = yaml.Unmarshal(b, &values); err != nil {
+					valuesMap := map[string]any{}
+					if err = yaml.Unmarshal(b, &valuesMap); err != nil {
 						return nil, err
 					}
+					values = helm.MergeMaps(values, valuesMap)
```

Comment:
Should we also merge the literal values using `helm.MergeMaps` rather than our own `mergeMaps` (see line 319)?   Perhaps use `helm.MergeMaps` when `AllowNullValues` is true (since the flag was intended to improve compatibility). Are there any differences in the logic in that case? Or we could leave it for the v4 implementation of `Release`.

Reply:
This is a good question. I added some test cases and I think this basically boils down to whether we also want to fix https://github.com/pulumi/pulumi-kubernetes/issues/2731 as part of this PR, because I think that's essentially what your suggestion would do.

I'm OK to leave line 319 as-is just to keep scope and risk down. I can follow up with #2731.